### ISSUE TITLE
The MSVC CRT library is not overrideable by other build systems

### ISSUE THREAD

**Issue by ChrisDenton** [rust-lang/rust#107570]
For `-windows-msvc` targets linking the C runtime (e.g. C main, etc) is done in the `libc` crate which then [hard codes](https://github.com/rust-lang/libc/blob/106b57a796145c42f9a385a2a8cf764abbc66624/src/windows/mod.rs#L258-L261) the CRT startup libraries to use, which means they can't be overridden (without hacks or `no_std`) except to choose between dynamic or static linking. This makes it difficult to select the right CRT for certain builds. This matters particularly when integrating Rust alongside existing C/C++ build systems. The same CRT must be used in all cases.

I would propose that `rustc` should behave more like `cl` (the MSVC compiler) in this case. We should use [`/DEFAULTLIB:`](https://learn.microsoft.com/en-us/cpp/build/reference/defaultlib-specify-default-library?view=msvc-170) to link the CRT so the library is overridable rather than hard coded. This also means it can work with existing tools rather than needing anything custom for `rustc`.

**Comment by spangaer** [comment]
It looks like you figured this out yourself?
https://github.com/rust-lang/rust/issues/39016#issuecomment-2391095973

Though I've noticed it doesn't seem to work with `1.79`, but does with `1.83`
https://github.com/rust-lang/rust/issues/39016#issuecomment-2521395154

**Comment by ChrisDenton** [comment]
Yes, this is an area I've been trying to improve. I made changes to CRT linking so it's now possible. I'll close this issue as completed.

However, it's not quite perfect. There's no way for other tools to know when CRT has been overridden, which they will need to know in order to build compatible C/C++ libraries.
